### PR TITLE
ssh: Ensure debug messages are properly converted to text

### LINKED ISSUE #57843: ssh: UnicodeDecodeError: 'ascii' codec can't decode byte
##### SUMMARY
Executing a playbook in verbose 3 (-vvv) leads to the following error:

> UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 63: ordinal not in range(128)
fatal: [server]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}

when following conditions are met:
- task in error: permission denied for example
- characters with an accent in connection messages: pre login message (file /etc/issue) for example

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ssh.py / display.py

##### ANSIBLE VERSION
```
ansible 2.6.13
  config file = /jenkins/test/ansible/ansible.cfg
  configured module search path = [u'/jenkins/test/ansible/lib/python2.6/site-packages/ansible/modules', u'/jenkins/test/ansible/odyssey/modules']
  ansible python module location = /jenkins/test/ansible/lib/python2.6/site-packages/ansible
  executable location = /jenkins/test/ansible/bin/ansible
  python version = 2.6.6 (r266:84292, May 22 2015, 08:34:51) [GCC 4.4.7 20120313 (Red Hat 4.4.7-15)]
```

##### CONFIGURATION
```
ANSIBLE_FORCE_COLOR(/jenkins/test/ansible/ansible.cfg) = True
ANSIBLE_PIPELINING(/jenkins/test/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/jenkins/test/ansible/ansible.cfg) = -o PreferredAuthentications=publickey
ANSIBLE_SSH_CONTROL_PATH(/jenkins/test/ansible/ansible.cfg) = %(directory)s/sge-ssh-%%h-%%p-%%r
ANSIBLE_SSH_RETRIES(/jenkins/test/ansible/ansible.cfg) = 0
ANY_ERRORS_FATAL(/jenkins/test/ansible/ansible.cfg) = True
CACHE_PLUGIN(/jenkins/test/ansible/ansible.cfg) = memory
DEFAULT_ACTION_PLUGIN_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible/project/modules/plugin_config_ssh/action_plugins']
DEFAULT_CALLBACK_PLUGIN_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible/project/roles/project_callback_plugins']
DEFAULT_CALLBACK_WHITELIST(/jenkins/test/ansible/ansible.cfg) = ['profile_tasks']
DEFAULT_FILTER_PLUGIN_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible/project/roles/project_filters_plugin']
DEFAULT_FORKS(/jenkins/test/ansible/ansible.cfg) = 50
DEFAULT_GATHERING(/jenkins/test/ansible/ansible.cfg) = explicit
DEFAULT_HASH_BEHAVIOUR(/jenkins/test/ansible/ansible.cfg) = merge
DEFAULT_JINJA2_EXTENSIONS(/jenkins/test/ansible/ansible.cfg) = jinja2.ext.do,jinja2.ext.i18n
DEFAULT_LOCAL_TMP(/jenkins/test/ansible/ansible.cfg) = /home/jenkins/.ansible/tmp/ansible-local-21303iN97VZ
DEFAULT_MANAGED_STR(/jenkins/test/ansible/ansible.cfg) = Ansible managed: {file} modified on %Y-%m-%d %H:%M:%S by {uid} on {host}
DEFAULT_MODULE_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible/lib/python2.6/site-packages/ansible/modules', u'/jenkins/test/ansible/project/modules']
DEFAULT_MODULE_UTILS_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible']
DEFAULT_POLL_INTERVAL(/jenkins/test/ansible/ansible.cfg) = 15
DEFAULT_REMOTE_PORT(/jenkins/test/ansible/ansible.cfg) = 22
DEFAULT_REMOTE_USER(/jenkins/test/ansible/ansible.cfg) = deploy
DEFAULT_ROLES_PATH(/jenkins/test/ansible/ansible.cfg) = [u'/jenkins/test/ansible/project/roles']
DEFAULT_SCP_IF_SSH(/jenkins/test/ansible/ansible.cfg) = True
DEFAULT_TIMEOUT(/jenkins/test/ansible/ansible.cfg) = 20
DEFAULT_TRANSPORT(/jenkins/test/ansible/ansible.cfg) = ssh
DISPLAY_SKIPPED_HOSTS(/jenkins/test/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/jenkins/test/ansible/ansible.cfg) = False
INVENTORY_UNPARSED_IS_FAILED(/jenkins/test/ansible/ansible.cfg) = True
SYSTEM_WARNINGS(/jenkins/test/ansible/ansible.cfg) = True
```

##### OS / ENVIRONMENT
target OS version: Red Hat Enterprise Linux Server release 7.3 (Maipo)

##### STEPS TO REPRODUCE
Playbook:
```yaml
- hosts: server
  tasks:
  - stat:
      path: "/home/anotheruser/lvi"
```
Set file /etc/issue on target machine with content:
```
ATTENTION:
Quiconque accéderait ou tenterait d'accéder à ce système.
```
Execute with ansible-playbook -vvv -i playbook.yml

Notes:
- path /home/anotheruser is not accessible to the current user
- running without -vvv (ie -vv or -v or none) does not lead to the error

##### EXPECTED RESULTS
Permission denied as /home/anotheruser is not accessible to the current user and as result is with ansible version < 2.6.13:
```
TASK [stat] *************************************************************************************************************************************************************
task path: /home/jenkins/playbook.yml:3
Friday 14 June 2019  10:26:01 +0200 (0:00:00.170)       0:00:00.170 ***********
Using module file /jenkins/test/ansible/lib/python2.6/site-packages/ansible/modules/files/stat.py
<server> ESTABLISH SSH CONNECTION FOR USER: myuser
<server> SSH: EXEC ssh -o PreferredAuthentications=publickey -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=myuser -o ConnectTimeout=20 server '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<server> (1, '\n{"msg": "Permission denied", "failed": true, "exception": "  File \\"/tmp/ansible_1CvBt6/ansible_module_stat.py\\", line 476, in main\\n    st = os.lstat(b_path)\\n", "invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "follow": false, "path": "/home/anotheruser/lvi", "get_md5": null, "get_mime": true, "get_attributes": true}}}\n', "ATTENTION:\nQuiconque acc\xc3\xa9derait ou tenterait d'acc\xc3\xa9der \xc3\xa0 ce syst\xc3\xa8me.\n\n")
The full traceback is:
  File "/tmp/ansible_1CvBt6/ansible_module_stat.py", line 476, in main
    st = os.lstat(b_path)

fatal: [server]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "checksum_algorithm": "sha1",
            "follow": false,
            "get_attributes": true,
            "get_checksum": true,
            "get_md5": null,
            "get_mime": true,
            "path": "/home/anotheruser/lvi"
        }
    },
    "msg": "Permission denied"
}
```

##### ACTUAL RESULTS
The execution failed with error "Unexpected failure during module execution":
```
TASK [stat] *************************************************************************************************************************************************************
task path: /home/jenkins/playbook.yml:3
Friday 14 June 2019  10:36:17 +0200 (0:00:00.170)       0:00:00.170 ***********
Using module file /jenkins/test/ansible/lib/python2.6/site-packages/ansible/modules/files/stat.py
<server> ESTABLISH SSH CONNECTION FOR USER: myuser
<server> SSH: EXEC ssh -o PreferredAuthentications=publickey -o StrictHostKeyChecking=no -o Port=22 -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=myuser -o ConnectTimeout=20 server '/bin/sh -c '"'"'/usr/bin/python && sleep 0'"'"''
<server> (1, '\n{"msg": "Permission denied", "failed": true, "exception": "  File \\"/tmp/ansible_Vvkdbf/ansible_module_stat.py\\", line 476, in main\\n    st = os.lstat(b_path)\\n", "invocation": {"module_args": {"checksum_algorithm": "sha1", "get_checksum": true, "follow": false, "path": "/home/anotheruser/lvi", "get_md5": null, "get_mime": true, "get_attributes": true}}}\n', "ATTENTION:\nQuiconque acc\xc3\xa9derait ou tenterait d'acc\xc3\xa9der \xc3\xa0 ce syst\xc3\xa8me.\n\n")
The full traceback is:
Traceback (most recent call last):
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/executor/task_executor.py", line 139, in run
    res = self._execute()
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/executor/task_executor.py", line 584, in _execute
    result = self._handler.run(task_vars=variables)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/action/normal.py", line 46, in run
    result = merge_hash(result, self._execute_module(task_vars=task_vars, wrap_async=wrap_async))
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/action/__init__.py", line 853, in _execute_module
    res = self._low_level_execute_command(cmd, sudoable=sudoable, in_data=in_data)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/action/__init__.py", line 960, in _low_level_execute_command
    rc, stdout, stderr = self._connection.exec_command(cmd, in_data=in_data, sudoable=sudoable)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/connection/ssh.py", line 1080, in exec_command
    (returncode, stdout, stderr) = self._run(cmd, in_data, sudoable=sudoable)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/connection/ssh.py", line 354, in wrapped
    _handle_error(remaining_retries, cmd[0], return_tuple, self._play_context.no_log, self.host)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/plugins/connection/ssh.py", line 307, in _handle_error
    display.vvv(msg, host=host)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/utils/display.py", line 183, in vvv
    return self.verbose(msg, host=host, caplevel=2)
  File "/jenkins/test/ansible/lib/python2.6/site-packages/ansible/utils/display.py", line 206, in verbose
    self.display("<%s> %s" % (host, msg), color=C.COLOR_VERBOSE, screen_only=True)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 63: ordinal not in range(128)

fatal: [server]: FAILED! => {
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}
```

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
ssh: Ensure debug messages are properly converted to text. Fixes #57843

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/plugins/connection/ssh.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/plugins/connection/ssh.py [hunk 1]
```diff
@@ -321,23 +321,23 @@ def _handle_error(remaining_retries, command, return_tuple, no_log, host, displa
...
+            msg = u'Invalid/incorrect username/password. Skipping remaining {0} retries to prevent account lockout:'.format(remaining_retries)
             if remaining_retries <= 0:
-                msg = 'Invalid/incorrect password:'
+                msg = u'Invalid/incorrect password:'
             if no_log:
-                msg = '{0} <error censored due to no log>'.format(msg)
+                msg = u'{0} <error censored due to no log>'.format(msg)
             else:
-                msg = '{0} {1}'.format(msg, to_native(return_tuple[2].rstrip()))
+                msg = u'{0} {1}'.format(msg, to_text(return_tuple[2]).rstrip())
```
Comment (sivel) [inline lib/ansible/plugins/connection/ssh.py]: Hrm, I might have been overzealous.  This is wrong.  This is used in an exception, not in display.

File: lib/ansible/plugins/connection/ssh.py [hunk 2]
```diff
@@ -835,12 +836,12 @@ def _bare_run(self, cmd, in_data, sudoable=True, checkrc=True):
                 # We're requesting escalation with a password, so we have to
                 # wait for a password prompt.
                 state = states.index('awaiting_prompt')
-                display.debug(u'Initial state: %s: %s' % (states[state], prompt))
-            elif self._play_context.become and self._play_context.success_key:
+                display.debug(u'Initial state: %s: %s' % (states[state], to_text(prompt)))
+            elif self._play_context.become and self.become.success:
```
Comment (sivel) [inline lib/ansible/plugins/connection/ssh.py:840]: This is somewhat unrelated, but I realized it was wrong when fixing the `display.debug` that comes next.

File: lib/ansible/plugins/connection/ssh.py [hunk 3]
```diff
@@ -356,11 +356,11 @@ def _handle_error(remaining_retries, command, return_tuple, no_log, host, displa
...
     # For other errors, no execption is raised so the connection is retried and we only log the messages
     if 1 <= return_tuple[0] <= 254:
-        msg = "Failed to connect to the host via ssh:"
+        msg = u"Failed to connect to the host via ssh:"
         if no_log:
-            msg = '{0} <error censored due to no log>'.format(msg)
+            msg = u'{0} <error censored due to no log>'.format(msg)
         else:
-            msg = '{0} {1}'.format(msg, to_native(return_tuple[2]).rstrip())
+            msg = u'{0} {1}'.format(msg, to_text(return_tuple[2]).rstrip())
```
Comment (sivel) [inline lib/ansible/plugins/connection/ssh.py:363]: This `msg` (and the 2 above in this block) leads to a display. All others lead to an exception.

**Review by bcoca — APPROVED** [review]
lgtm, don't think you needed to `u''` everything, but it doesn't hurt either.

**Pull request merged**